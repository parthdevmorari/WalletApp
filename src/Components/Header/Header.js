import React, { Component } from "react";
import { View, Text, } from "react-native";
import styles from "./HeaderStyle";

const Header = (props) => {

	const { title, containerStyle, } = props;

		return(
			<View style={[styles.homeContainer,containerStyle ]} >
                <View style={styles.headerLeft}>
				<View style={styles.homeLogo}>
		 		<Text style={styles.homeText} >{title}</Text>		
			 </View>
                </View>
                
               
            </View>
		)
	
}

export default Header;

