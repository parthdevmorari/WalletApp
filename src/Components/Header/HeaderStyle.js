import { StyleSheet, StatusBar } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { color } from '../../../Constants/Color'
import { theme } from '../../../Constants/Theme'
const styles = StyleSheet.create({

	homeContainer: {
		backgroundColor: theme.THEME_COLOR,
		width: '100%',
		paddingHorizontal: 15,
		paddingTop: StatusBar.currentHeight,
		flexDirection: 'row',
		justifyContent: 'space-between',
		shadowColor: "#000",
		shadowOffset: {
			width: 0,
			height: 2,
		},
		shadowOpacity: 0.25,
		shadowRadius: 3.84,
		borderBottomWidth: 2,
		borderBottomColor: theme.THEME_COLOR_1,
	},

	headerLeft: {
		flex: 0.3,
		justifyContent: 'center',
		alignItems: 'flex-start',

	},
	homeLogo: {
		borderRadius: 10,
		height: 60,
		width: 150,
		marginLeft: 5,

	},
	homeText: {
		fontWeight: '700',
		fontSize: wp(6.5),
		color: color._WHITE,
		textAlignVertical: 'bottom'
	},




});
export default styles;