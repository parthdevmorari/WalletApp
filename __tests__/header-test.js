import 'react-native';
import React from 'react';
import Header from '../src/Components/Header/Header'
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

describe('Header Component', () => {
  test('Header Component', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.exists()).toBe(true);
  });
  it('Header  toMatchSnapshot', () => {
    const wrapper = shallow(<Header />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});