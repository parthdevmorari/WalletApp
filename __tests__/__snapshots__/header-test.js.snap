// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Header  toMatchSnapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#03386F",
        "borderBottomColor": "#ddd",
        "borderBottomWidth": 2,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 15,
        "paddingTop": null,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 0.3,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 10,
          "height": 60,
          "marginLeft": 5,
          "width": 150,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 49,
            "fontWeight": "700",
            "textAlignVertical": "bottom",
          }
        }
      />
    </View>
  </View>
</View>
`;
